test/Test.hs

```diff
@@ -1,7 +1,191 @@
+data TestF a = TestF !Int [a]
+  deriving (Show, Eq, Generic, Functor)
+
+instance Eq1 TestF where
+  liftEq eq (TestF i1 xs1) (TestF i2 xs2)
+    = i1 == i2 && liftEq eq xs1 xs2
+
+instance Show1 TestF where
+  liftShowsPrec sp sl d (TestF i xs) =
+    showsBinaryWith showsPrec (const sl) "TestF" d i xs
+
+instance Fun.Arg a => Fun.Arg (TestF a)
+
+
+-- Note: size
+--
+-- I've found it quite difficult to get a good range of
+-- sizes for SeqT TestM Int using the basic tools in
+-- Gen. Preventing almost all examples being tiny seems to lead to
+-- some examples being unmanageably enormous. So I've decided to
+-- go with a "nuclear option". First, I select the exact total number
+-- of nodes in the SeqT. Then at each stage, the exact total size is
+-- chosen in advance to make sure the target is met.
+
+-- | Generate a partition of a non-negative integer into positive
+-- integers. This is not statistically fair because I'm not that smart.
+splat :: MonadGen m => Size -> m [Size]
+splat 0 = pure []
+splat n = do
+  k <- Gen.integral (Range.constant 1 n)
+  rest <- splat (n - k)
+  pure (k : rest)
+
+genTestFSized :: MonadGen m => (Size -> m a) -> Size -> m (TestF a)
+genTestFSized m sz = do
+  i <- Gen.integral (Range.constant 1 10000)
+  part <- splat sz
+  goop <- traverse m part
+  pure (TestF i goop)
+
+newtype TestM a = TestM (Free TestF a)
+  deriving (Show, Eq, Generic, Functor, Applicative, Monad, Eq1)
+
+instance Show1 TestM where
+  liftShowsPrec sp sl d (TestM m) = showsUnaryWith (liftShowsPrec sp sl) "TestM" d m
+
+genTestMSized :: MonadGen m => (Size -> m a) -> Size -> m (TestM a)
+genTestMSized = \m sz -> TestM <$> go m sz
+  where
+    go :: MonadGen m => (Size -> m a) -> Size -> m (Free TestF a)
+    go m n | n <= 1 = Pure <$> m (n - 1)
+    go m n = Free <$> genTestFSized (go m) (n - 1)
+
+-- | Generate a test monad value.
+genTestM :: MonadGen m => m a -> m (TestM a)
+genTestM m = Gen.sized $ \sz -> do
+  true_size <- Gen.integral (Range.constant 0 sz)
+  genTestMSized (const m) true_size
+
+simpleTestM :: MonadGen m => m (TestM Int)
+simpleTestM = genTestM (Gen.integral $ Range.constant 0 5)
+
+listToQueue :: [a] -> Queue a
+listToQueue = foldl' (S.|>) S.empty
+
+genViewSized :: forall m a. MonadGen m => m a -> Size -> m (View TestM a)
+genViewSized _ sz | sz <= 1 = pure Empty
+genViewSized m sz = do
+  a <- m
+  s <- genSeqTSized m (sz - 1)
+  pure (a :< s)
+
+genSeqTSized :: forall m a. MonadGen m => m a -> Size -> m (SeqT TestM a)
+genSeqTSized m sz = do
+  part <- splat sz
+  goop <- traverse (genTestMSized (genViewSized m)) part
+  pure $ SeqT $ listToQueue goop
+
+genSeqT :: forall m a. MonadGen m => m a -> m (SeqT TestM a)
+genSeqT m = Gen.sized $ \sz -> do
+  tsz <- Gen.integral (Range.linear 0 sz)
+  genSeqTSized m tsz
+  
+simpleSeqT :: MonadGen m => m (SeqT TestM Int)
+simpleSeqT = genSeqT (Gen.integral $ Range.constant 0 5)
+
+simpleSeqT' :: MonadGen m => m (Size, SeqT TestM Int)
```

Comment:
I don't think this is used anywhere. I just put it together to debug my test case size issues.